client/v2_2/docker_image_.py

```diff
@@ -337,6 +337,11 @@ def _content(self, name, memoize=True):
 
   def _gzipped_content(self, name):
     """Returns the result of _content with gzip applied."""
+    # Allow the legacy tarball to contain pre-compressed versions of the layers.
+    try:
+      return self._content(name + '.gz', memoize=False)
```

Comment:
I'm a little concerned that this will add an additional (fruitless) seek through the tarball in the common case where it doesn't have the compressed tarball.